.github/workflows/mc-ledger-test.yml

```diff
@@ -0,0 +1,98 @@
+name: mc-ledger-test
+
+env:
+  SGX_MODE: SW
+  IAS_MODE: DEV
+  RUST_BACKTRACE: full
+  IAS_API_KEY: ${{ secrets.SGX_PRIMARY_API_KEY }}
+  IAS_SPID: ${{ secrets.SGX_SPID }}
+  MC_LOG: warn
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_a_ref:
+        description: 'Tag or commit-hash for the base version of Full-Service'
+        type: string
+        required: true
+        default: 'v1.9.0' 
+      version_b_ref:
+        description: 'Tag or commit-hash to upgrade Full-Service to'
+        type: string
+        required: true
+        default: "v1.9.2"
+  push:
+    branches:
+      - setup-mobilecoin
+      - pouneh/e2e_test_infra
+    inputs: 
+      version_a_ref:
+        description: 'Tag or commit-hash for the base version of Full-Service'
+        required: true
+        default: 'v1.10.0-pre.5' #'v1.9.0' 
+      version_b_ref:
+        description: 'Tag or commit-hash to upgrade Full-Service to'
+        required: true
+        default: 'v1.9.2 '
+
+# TESTING WITH NEKTOS/ACT
+# nektos/act is a 3rd party github library that allows for the LOCAL testing of github actions
+# best nektox/act testing flow is:
+  # 2 change the runs-on tag to point to `ubuntu-latest` if its set to a self-hosted tag
+    # 2.1 `ubuntu-latest` will use Github Action minutes, but is required for testing with nektos/act
+    # 2.2 `[self-hosted, Linux, large]` will use a self-hosted machine (free for us), but fails with nektos/act
```

Comment:
FYI `act` has a `-P` flag to override the platform mappings; I usually pass `-P self-hosted=mobilecoin/builder-install:latest` or `-P self-hosted=ubuntu-latest`